cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -52,12 +54,84 @@
 #include <memory>
 #include <optional>
 #include <type_traits>
+#include <unordered_set>
 #include <utility>
+#include <variant>
 
 namespace CUDF_EXPORT cudf {
 
 namespace experimental {
 
+/// Type identity type from C++20
+template <typename T>
+using type_identity_t = T;
+
+/**
+ * @brief Recursive template to apply type transformations sequentially. Transformations are
+ * applied first-to-last in the order specified.
+ *
+ * @tparam T Base type or initial type on which transformations are applied
+ * @tparam Rest Type transformations to apply
+ */
+template <typename T, template <typename> typename... Rest>
+struct transform_sequence;
```

Comment:
Depending on how other reviewers see this, `transform` is reserved for the use of algorithms in C++ and the structs here are basically dispatch helpers.